### DISCUSSION TITLE
ACL module issues

### DISCUSSION THREAD

**Discussion by Danieleeffe1** [OpenMage/magento-lts#3417] [Extensions & Modules]
years ago I bought an Access Control Level module for magento 1.9.x community edition.

the module has always worked but now with the latest release of OpenMage there is a strange bug.

when the ACL module is installed on the front page, the quicksearch module and the default links in the footer disappear.

the manufacturer of the module has already told me that he is not interested in giving me support unless he spends more than the module itself.

at this point I wonder if the community is interested in taking a cue from this module (which I will make available) to create one integrated into openmage.

I suspect the problem is here

[Config.zip](https://github.com/OpenMage/magento-lts/files/12206724/Config.zip)

**Comment by fballiano** [comment]
What are the features that this module add to the core?

**Reply by Danieleeffe1** [reply]
the file is install in
App/core/local/mage/core/model/config.php 

I have this module (see the link)

[https://www.safemage.com/advanced-permissions-manager.html](https://www.safemage.com/advanced-permissions-manager.html)

**Reply by sreichel** [reply]
Just for notice ... I think there is an open issue to add missing permission for all pages and actions.

**Reply by fballiano** [reply]
@Danieleeffe1 I was thinking we could add these functionalities to the core but it seems a quite big module to replace, not an easy task :-\

**Comment by kiatng** [comment]
Q1. The safemage extension seems to be for a store with lots of admin users, whom probably do not work for free. If it is important to have advanced ACL for a store with significant running costs, why is it that you do not engage a coder to fix the issue?

Q2. ACL is for backend. If it affects frontend, it's not doing what it is expected. Does the extension has any frontend layout update or frontend blocks? 
1. Can you search for "top.search" for the search in the header, search for "removeLink" or "removeLinkByUrl" to check if the `catalogsearch` links were removed. 
2. You should turn on _Template Path Hints_ and _Add Block Names to Hints_ and examine the code.

**Comment by addison74** [comment]
1 - I am sorry to say it publicly, but do you really think that someone can solve the problems of that extension just based on a posted file?

2 - The mentioned extension is fine and would be useful in OpenMage, especially when you have several administrators and you don't want them to intervene on content that doesn't belong to them. A code cannot be integrated without the consent of the person who created it and who owns all the rights to it. Even if that developer no longer wants to work on it, nothing more can be done, You have to hire someone to solve your issue, but it cannot be integrated into OpenMage.

3 - ACL only works in Backend as @kiatng said previously. Something else creates problems so that those blocks are not displayed in the Frontend. You should do an installation from scratch in a test environment and only using this extension, only then you can draw conclusions.

**Comment by Danieleeffe1** [comment]
Sorry for the delay in responses.

Do any of you want to see the module to solve the problem and use it as a basis to understand how to reproduce it in OpenMage in a legal way?
